File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//find letters per 100 words
//find sentences per 100 words
int main(void)
{
    string text = get_string("text: ");
    int sentencecount = 0;
    int wordcount = 1;
    int lettercount = 0;
    
    for (int a = 0; a < strlen(text); a++)
    {
        if ((text[a] >= 'a' && text[a] <= 'z') || (text[a] >= 'A' && text[a] <= 'Z'))
        {
            lettercount++;
        }
        
        if (text[a] == ' ')
        {
            wordcount++;
        }
        if (text[a] == '.' || text[a] == '?' || text[a] == '!')
        {
            sentencecount++;
        }
    }

    float lpw = ((float)lettercount/(float)wordcount) * 100;
    float spw = ((float)sentencecount/(float)wordcount) * 100;
    
    float index = round(0.0588 * lpw - 0.296 * spw - 15.8);
    int w = index;
    
    if(w < 0){
        printf("Before Grade 1...\n");
    }
    else if (w > 16){
        printf("Grade 16+\n");
    }
    else printf("Grade: %d\n", w);

}

File: readability.py
from cs50 import get_string

text = get_string("text: ")
SentCount = 0
WordCount = 1
LettCount = 0
strlen = len(text)
for a in range(strlen):
    if((text[a] >= 'a' and text[a] <= 'z') or (text[a] >= 'A' and text[a] <= 'Z')):
        LettCount += 1
    if (text[a] == ' '):
        WordCount += 1
    if (text[a] == '.' or text[a] == '?' or text[a] == '!'):
        SentCount += 1
lpw = float((LettCount / WordCount) * 100)
spw = float((SentCount / WordCount) * 100)
index = float(round(0.0588 * lpw - 0.296 * spw - 15.8))
w = int(index)
if (w < 0):
    print("Before Grade 1...")
elif (w > 16):
    print("Grade 16+")
else:
    print(f"Grade {w}")